src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/EventAuditRepository.kt

```diff
@@ -0,0 +1,31 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.repository
+
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Modifying
+import org.springframework.data.jpa.repository.Query
+import org.springframework.stereotype.Repository
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.visitscheduler.model.ApplicationMethodType
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.EventAudit
+
+@Repository
+interface EventAuditRepository : JpaRepository<EventAudit, Long> {
+  fun findByBookingReference(bookingReference: String): List<EventAudit>
+
+  @Query(
+    "SELECT * FROM event_audit " +
+      "WHERE booking_reference = :bookingReference AND type = 'BOOKED_VISIT'  " +
```

Comment:
should 'BOOKED_VISIT' be replaced with enum name?

Reply:
I suppose si